presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -363,7 +363,7 @@ public static HiveColumnStatistics fromMetastoreApiColumnStatistics(ColumnStatis
             BooleanColumnStatsData booleanStatsData = columnStatistics.getStatsData().getBooleanStats();
             return createBooleanColumnStatistics(
                     booleanStatsData.isSetNumTrues() ? OptionalLong.of(booleanStatsData.getNumTrues()) : OptionalLong.empty(),
```

Comment:
since num trues is also hardcoded in impala, when numfalse is set to -1, you should also return empty stats for num trues.

Reply:
@rschlussel  I have modified the code and re-submitted;